### PR TITLE
[2.3][Translation] test refresh cache when resources File change.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | ~ |
| Tests pass? | yes |
| License | MIT |

For 2.6 https://github.com/symfony/symfony/pull/14280

### CODE REVIEW

**Comment by aitboudad** [comment]
@mpdude I made an update that refer to your comment, is it ok :) ?

**Comment by mpdude** [comment]
:+1: for the "expect never" change.

Besides that, I wouldn't touch the cache file. How that works (and what effects it has) is an implementation detail of the `ConfigCache`. And I don't see what additional assertions this change would bring.

Maybe someone else can comment?

**Comment by aitboudad** [comment]
it seems to be OK, I'll merge it.

**Comment by mpdude** [comment]
Disregard my previous comment – I was looking at an outdated version of the "Files changed" tab. 

Sorry.

**Comment by xabbuh** [comment]
@aitboudad It seems that this PR makes builds for the [low deps fail on higher branches](https://travis-ci.org/symfony/symfony/jobs/58288946#L3255). Do you have an idea what's going wrong there?

File: src/Symfony/Bundle/FrameworkBundle/Tests/Translation/TranslatorTest.php [hunk 1]
```diff
@@ -91,6 +93,27 @@ public function testTransWithCaching()
...
+
+        // refresh cache again when resource file resources file change
+        $resource = $this->getMock('Symfony\Component\Config\Resource\ResourceInterface');
+        $resource
+            ->expects($this->at(0))
+            ->method('isFresh')
+            ->will($this->returnValue(false))
+        ;
+        $catalogue = $this->getCatalogue('fr', array('foo' => 'foo fresh'));
+        $catalogue->addResource($resource);
```
Comment (mpdude) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Translation/TranslatorTest.php]: @xabbuh haven't looked at it in detail, but I think this resource mock has no effect. Instead the resource already serialized in the cache will be loaded (and still be fresh)?

**Comment by mpdude** [comment]
Have a look at the testCatalogueIsReloadedWhenResourcesAreNoLongerFresh() Method in #14291 where I do something similar, maybe that helps?

**Comment by aitboudad** [comment]
I'll look into it :)

**Comment by aitboudad** [comment]
fixed in #14344

**Pull request merged**